gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslator.java

```diff
@@ -138,6 +139,11 @@ protected String getSyntax(final Date o) {
             return "time.UnixMilli(" + o.getTime() + ")";
         }
 
+        @Override
+        protected String getSyntax(final OffsetDateTime o) {
+            return "time.UnixMilli(" + o.toInstant().toEpochMilli() + ")";
```

Comment:
We may be able to preserve more precision as well as the timezone offset via https://pkg.go.dev/time#Date